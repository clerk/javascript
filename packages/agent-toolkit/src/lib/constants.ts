import { clerkClient } from './clerk-client';
import type { CreateClerkToolkitParams } from './types';

export const defaultCreateClerkToolkitParams = {
  allowPrivateMetadata: false,
  clerkClient,
} satisfies CreateClerkToolkitParams;
