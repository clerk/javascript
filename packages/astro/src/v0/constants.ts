const apiKey = import.meta.env.CLERK_API_KEY || '';

const secretKey = import.meta.env.CLERK_SECRET_KEY || '';

const apiVersion = import.meta.env.CLERK_API_VERSION || 'v1';

const apiUrl = import.meta.env.CLERK_API_URL || 'https://api.clerk.dev';

const frontendApi = import.meta.env.PUBLIC_ASTRO_APP_CLERK_FRONTEND_API || '';

const publishableKey = import.meta.env.PUBLIC_ASTRO_APP_CLERK_PUBLISHABLE_KEY || '';

const signInUrl = import.meta.env.PUBLIC_ASTRO_APP_CLERK_SIGN_IN_URL || ('' as string);

const signUpUrl = import.meta.env.PUBLIC_ASTRO_APP_CLERK_SIGN_UP_URL || ('' as string);

const jwtKey = import.meta.env.CLERK_JWT_KEY || '';

const PUBLISHABLE_KEY = publishableKey;
const SECRET_KEY = secretKey;
const SIGN_IN_URL = signInUrl;
const SIGN_UP_URL = signUpUrl;
const DOMAIN = import.meta.env.PUBLIC_ASTRO_APP_CLERK_DOMAIN || ('' as string);
const PROXY_URL = import.meta.env.PUBLIC_ASTRO_APP_CLERK_PROXY_URL || ('' as string);
const IS_SATELLITE = import.meta.env.PUBLIC_ASTRO_APP_CLERK_IS_SATELLITE === 'true';
const API_VERSION = import.meta.env.CLERK_API_VERSION || 'v1';
const API_URL = import.meta.env.CLERK_API_URL || 'https://api.clerk.dev';

const CLERK_JS_URL = import.meta.env.PUBLIC_ASTRO_APP_CLERK_JS_URL;
const CLERK_JS_VARIANT = import.meta.env.PUBLIC_ASTRO_APP_CLERK_JS_VARIANT;
const CLERK_JS_VERSION = import.meta.env.PUBLIC_ASTRO_APP_CLERK_JS_VERSION;

export {
  secretKey,
  apiKey,
  apiUrl,
  apiVersion,
  frontendApi,
  publishableKey,
  jwtKey,
  signInUrl,
  signUpUrl,
  PUBLISHABLE_KEY,
  SECRET_KEY,
  SIGN_IN_URL,
  SIGN_UP_URL,
  DOMAIN,
  PROXY_URL,
  IS_SATELLITE,
  API_URL,
  API_VERSION,
  CLERK_JS_URL,
  CLERK_JS_VARIANT,
  CLERK_JS_VERSION,
};
