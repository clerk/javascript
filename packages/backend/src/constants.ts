export const API_URL = 'https://api.clerk.dev';
export const API_VERSION = 'v1';

// TODO: Get information from package.json or define them from ESBuild
export const USER_AGENT = `@clerk/backend`;
export const MAX_CACHE_LAST_UPDATED_AT_SECONDS = 5 * 60;

const Attributes = {
  AuthStatus: '__clerkAuthStatus',
  AuthReason: '__clerkAuthReason',
  AuthMessage: '__clerkAuthMessage',
} as const;

const Cookies = {
  Session: '__session',
  ClientUat: '__client_uat',
} as const;

const Headers = {
  AuthStatus: 'x-clerk-auth-status',
  AuthReason: 'x-clerk-auth-reason',
  AuthMessage: 'x-clerk-auth-message',
  EnableDebug: 'x-clerk-debug',
  ClerkRedirectTo: 'x-clerk-redirect-to',
  Authorization: 'authorization',
  ForwardedPort: 'x-forwarded-port',
  ForwardedProto: 'x-forwarded-proto',
  ForwardedHost: 'x-forwarded-host',
  Referrer: 'referer',
  UserAgent: 'user-agent',
  Origin: 'origin',
  Host: 'host',
  ContentType: 'content-type',
} as const;

const SearchParams = {
  AuthStatus: Headers.AuthStatus,
} as const;

const ContentTypes = {
  Json: 'application/json',
} as const;

export const constants = {
  Attributes,
  Cookies,
  Headers,
  SearchParams,
  ContentTypes,
} as const;
