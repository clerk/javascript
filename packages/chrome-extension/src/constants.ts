export const CLIENT_JWT_KEY = '__client';
export const STORAGE_KEY_CLIENT_JWT = '__clerk_client_jwt';
export const VALID_HOST_PERMISSION_REGEX = /(https?:\/\/[\w.-]+)/;
export const DEFAULT_LOCAL_HOST_PERMISSION = 'http://localhost';
