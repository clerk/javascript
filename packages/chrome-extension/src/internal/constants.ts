export const CLIENT_JWT_KEY = '__client';
export const STORAGE_KEY_CLIENT_JWT = '__clerk_client_jwt';
export const DEFAULT_LOCAL_HOST_PERMISSION = 'http://localhost';
export const AUTH_HEADER = {
  production: 'Authorization',
  development: 'Clerk-Db-Jwt',
};
