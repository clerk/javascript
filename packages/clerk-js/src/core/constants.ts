export const PRESERVED_QUERYSTRING_PARAMS = ['after_sign_in_url', 'after_sign_up_url', 'redirect_url'];

export const DEV_BROWSER_SSO_JWT_KEY = 'clerk-db-jwt';
export const DEV_BROWSER_SSO_JWT_PARAMETER = '__dev_session';
export const DEV_BROWSER_SSO_JWT_HTTP_HEADER = 'Clerk-Cookie';

export const CLERK_MODAL_STATE = '__clerk_modal_state';
/**
 * @deprecated This will be removed in the next minor version
 */
export const CLERK_REFERRER_PRIMARY = '__clerk_referrer_primary';
export const CLERK_SYNCED = '__clerk_synced';
export const CLERK_SATELLITE_URL = '__clerk_satellite_url';
export const ERROR_CODES = {
  FORM_IDENTIFIER_NOT_FOUND: 'form_identifier_not_found',
  FORM_PASSWORD_INCORRECT: 'form_password_incorrect',
  INVALID_STRATEGY_FOR_USER: 'strategy_for_user_invalid',
  NOT_ALLOWED_TO_SIGN_UP: 'not_allowed_to_sign_up',
  OAUTH_ACCESS_DENIED: 'oauth_access_denied',
  OAUTH_EMAIL_DOMAIN_RESERVED_BY_SAML: 'oauth_email_domain_reserved_by_saml',
  NOT_ALLOWED_ACCESS: 'not_allowed_access',
  SAML_USER_ATTRIBUTE_MISSING: 'saml_user_attribute_missing',
};
