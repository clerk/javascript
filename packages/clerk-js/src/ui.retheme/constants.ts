export const USER_PROFILE_NAVBAR_ROUTE_ID = {
  ACCOUNT: 'account',
  SECURITY: 'security',
};

export const ORGANIZATION_PROFILE_NAVBAR_ROUTE_ID = {
  MEMBERS: 'members',
  SETTINGS: 'settings',
};
