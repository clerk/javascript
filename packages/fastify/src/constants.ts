import { constants } from '@clerk/backend';

export const API_URL = process.env.CLERK_API_URL || 'https://api.clerk.dev';
export const API_VERSION = process.env.CLERK_API_VERSION || 'v1';
/**
 * Backend API key
 * @deprecated Use `CLERK_SECRET_KEY` instead.
 */
export const API_KEY = process.env.CLERK_API_KEY || '';
export const SECRET_KEY = process.env.CLERK_SECRET_KEY || '';
/**
 * @deprecated Use `CLERK_PUBLISHABLE_KEY` instead.
 */
export const FRONTEND_API = process.env.CLERK_FRONTEND_API || '';
export const PUBLISHABLE_KEY = process.env.CLERK_PUBLISHABLE_KEY || '';
export const JWT_KEY = process.env.CLERK_JWT_KEY || '';

export const { Cookies, Headers } = constants;
