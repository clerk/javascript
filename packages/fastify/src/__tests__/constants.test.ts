const clonedEnvVars = {
  CLERK_API_URL: process.env.CLERK_API_URL,
  CLERK_API_VERSION: process.env.CLERK_API_VERSION,
  CLERK_SECRET_KEY: process.env.CLERK_SECRET_KEY,
  CLERK_PUBLISHABLE_KEY: process.env.CLERK_PUBLISHABLE_KEY,
  CLERK_JWT_KEY: process.env.CLERK_JWT_KEY,
};

process.env.CLERK_API_URL = 'CLERK_API_URL';
process.env.CLERK_API_VERSION = 'CLERK_API_VERSION';
process.env.CLERK_SECRET_KEY = 'CLERK_SECRET_KEY';
process.env.CLERK_PUBLISHABLE_KEY = 'CLERK_PUBLISHABLE_KEY';
process.env.CLERK_JWT_KEY = 'CLERK_JWT_KEY';

import * as constants from '../constants';

describe('constants', () => {
  afterEach(() => {
    Object.assign(process.env, clonedEnvVars);
  });

  test('from environment variables', () => {
    jest.resetModules();
    const { Headers, Cookies, ...localConstants } = constants;

    // Verify imported constants exist but don't snapshot them
    expect(Headers).toBeDefined();
    expect(Cookies).toBeDefined();

    // Only snapshot our local constants
    expect(localConstants).toMatchSnapshot();
  });
});
