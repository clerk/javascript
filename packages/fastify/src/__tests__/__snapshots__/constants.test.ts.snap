// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`constants from environment variables 1`] = `
{
  "API_URL": "CLERK_API_URL",
  "API_VERSION": "CLERK_API_VERSION",
  "Cookies": {
    "ClientUat": "__client_uat",
    "DevBrowser": "__clerk_db_jwt",
    "Handshake": "__clerk_handshake",
    "Session": "__session",
  },
  "Headers": {
    "Accept": "accept",
    "AuthMessage": "x-clerk-auth-message",
    "AuthReason": "x-clerk-auth-reason",
    "AuthSignature": "x-clerk-auth-signature",
    "AuthStatus": "x-clerk-auth-status",
    "AuthToken": "x-clerk-auth-token",
    "Authorization": "authorization",
    "ClerkRedirectTo": "x-clerk-redirect-to",
    "ClerkRequestData": "x-clerk-request-data",
    "ClerkUrl": "x-clerk-clerk-url",
    "CloudFrontForwardedProto": "cloudfront-forwarded-proto",
    "ContentType": "content-type",
    "EnableDebug": "x-clerk-debug",
    "ForwardedHost": "x-forwarded-host",
    "ForwardedPort": "x-forwarded-port",
    "ForwardedProto": "x-forwarded-proto",
    "Host": "host",
    "Location": "location",
    "Origin": "origin",
    "Referrer": "referer",
    "SecFetchDest": "sec-fetch-dest",
    "UserAgent": "user-agent",
  },
  "JWT_KEY": "CLERK_JWT_KEY",
  "PUBLISHABLE_KEY": "CLERK_PUBLISHABLE_KEY",
  "SDK_METADATA": {
    "environment": "test",
    "name": "@clerk/fastify",
    "version": "0.0.0-test",
  },
  "SECRET_KEY": "CLERK_SECRET_KEY",
}
`;
